File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int n=s.length();
        int m=0;
        if(n==0){
            return 0;
        }
        bool c[128]={false};
        for(int i=0;i<n;i++){
            if(i>0){
                for(int k=0;k<128;k++){
                    c[k]=false;
                }
            }
            int j=i;
            while(j<n&&!c[s[j]]){
                c[s[j]]=true;
                m=max(m,j-i+1);
                j++;
            }
        }
        return m;
    }
};

File: 0003-longest-substring-without-repeating-characters/0003-longest-substring-without-repeating-characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        char={}
        l=0
        mx=0
        for r in range(len(s)):
            curr=s[r]
            if curr in char:
                if char[curr]>=l:
                    l=char[curr]+1
            char[curr]=r
            curr=r-l+1
            if curr>mx:
                mx=curr
        return mx